### PR TITLE
Bugfix FXIOS-9221 [Multi-window] Sync zoom level across multiple windows

### PR DESCRIPTION
**Pull request by mattreaganmozilla**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9221)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/20434)

## :bulb: Description

Fixes zoom levels so that they work across multiple windows when the same host is on the selected tab.

### Video

https://github.com/mozilla-mobile/firefox-ios/assets/145381717/f8d4cabd-7497-45a1-add9-e1a0e0d5efbc

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Library/Downloads/DownloadsPanel.swift [hunk 1]
```diff
@@ -7,11 +7,6 @@ import Shared
 import Storage
 import Common
 
-extension Notification.Name {
-    // General notification posted when a file has been deleted from the downloads manager
-    public static let DownloadPanelFileWasDeleted = Notification.Name("DownloadPanelFileWasDeleted")
-}
-
```
Comment (mattreaganmozilla) [inline firefox-ios/Client/Frontend/Library/Downloads/DownloadsPanel.swift:14]: (Note: general cleanup, unrelated to this specific fix.)

File: firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift [hunk 2]
```diff
@@ -7,12 +7,6 @@ import Shared
 import ComponentLibrary
 import Common
 
-extension Notification.Name {
-    /// General notification posted when the default search engine has changed
-    public static let SearchSettingsDidUpdateDefaultSearchEngine =
-    Notification.Name("SearchSettingsDidUpdateDefaultSearchEngine")
-}
-
```
Comment (mattreaganmozilla) [inline firefox-ios/Client/Frontend/Settings/SearchSettingsTableViewController.swift:15]: (Note: general cleanup, unrelated to this specific fix.)

**Comment by mattreaganmozilla** [comment]
@PARAIPAN9 I assigned to you since I believe you've done the most work on the zoom bar?

**Review by PARAIPAN9 — APPROVED** [review]
Great Job!! Didn't know about `caseInsensitiveCompare`.  I've always used `lowercased()` to compare strings, but I made some investigation and `caseInsensitiveCompare` could be better in performance for long strings given the fact that  it can stop as soon as it finds a difference between the two strings.

**Comment by mattreaganmozilla** [comment]
@PARAIPAN9 Thank you for the prompt review, going to merge for now but if any additional feedback comes up later I can address in a follow-up.

**Pull request merged**